### PR TITLE
fmi2: add new recipe

### PR DESCRIPTION
**Pull request by joakimono**
Specify library name and version:  **fmi2/2.0.4**

This adds major version 2 of fmi from https://fmi-standard.org/, where the rationale is that fmi v1, v2, v3 can be required at the same time in downstream dependencies. A related PR for fmi1 is #19833. 

This is a dependency of a new package to be added: fmilibrary

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/fmi2/all/conandata.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+sources:
+  "2.0.4":
+    url:
+      - https://github.com/modelica/fmi-standard/releases/download/v2.0.4/FMI-Standard-2.0.4.zip
+    sha256: "79abb211052d28354a0efff0c6b1998b60a973dd75fdcb338db99ae1b970c9b2"
+patches:
+  "2.0.4": []
```
Comment (RubenRBS) [inline recipes/fmi2/all/conandata.yml:7]: If no patches, this section can be removed :)

```suggestion
```

File: recipes/fmi2/all/test_package/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,7 @@
+cmake_minimum_required(VERSION 3.15)
+project(test_package LANGUAGES C CXX)
+
+find_package(fmi2 REQUIRED CONFIG)
+
+add_executable(${PROJECT_NAME} test_package.cpp)
+target_link_libraries(${PROJECT_NAME} PRIVATE fmi2::fmi2)
```
Comment (RubenRBS) [inline recipes/fmi2/all/test_package/CMakeLists.txt:7]: Quick question, what's the minimum cppstd for these libraries?

**Review by RubenRBS — DISMISSED** [review]
Thanks!

**Review by RubenRBS — APPROVED** [review]

File: recipes/fmi2/all/test_package/CMakeLists.txt (hunk 2 above)
Comment (joakimono) [reply recipes/fmi2/all/test_package/CMakeLists.txt:7]: These are C headers, so I guess it does not apply.

Comment (RubenRBS) [reply recipes/fmi2/all/test_package/CMakeLists.txt:7]: Oh, hadn't realized that, thanks for the extra info :)

**Pull request merged**